File: lib/emerald/nodes/nested.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

require 'treetop'
require_relative 'Node'

#
# Appends compiled html for nested rule to the output.
# elements[0] is the tag statement, elements[4] is any nested html.
#
class Nested < Node
  def to_html(context)
    elements[0].opening_tag(context) +
      elements[4].to_html(context) +
      elements[0].closing_tag(context)
  end
end

File: cpp/src/nodes/nested.cpp
#include "nested.hpp"

Nested::Nested(NodePtr tag_statement, NodePtr root) : tag_statement(tag_statement), root(root) {}

std::string Nested::to_html(Json &context) {
  return tag_statement->opening_tag() + root->to_html(context) + tag_statement->closing_tag();
}
